src/server.js

```diff
@@ -0,0 +1,67 @@
+const { writeFile: writeFileAsync } = require("fs");
+const { promisify } = require("util");
+const logger = require("./tools/logger");
+const config = require("./config/config");
+
+const writeFile = promisify(writeFileAsync);
+
+let app;
+let server;
+let reload = Promise.resolve();
+
+// Launch Node.js server
+function launch(callback) {
+  app = require("./app"); // eslint-disable-line global-require
+  app.set("port", config.app.port);
+  server = app.listen(app.get("port"), () => {
+    logger.info(
+      `The server is running at http://localhost:${app.get("port")}/`
+    );
+    if (callback) callback();
+  });
+}
+
+// Shutdown Node.js server and database clients
+async function shutDown() {
+  const stats = app.get("stats");
+  if (stats) {
+    await writeFile("stats.txt", stats.join("\n"), "utf8");
+  }
+
+  if (server) {
+    await new Promise(resolve =>
+      server.close(() => {
+        resolve();
+      })
+    );
+  }
+}
+
+function handleError(err) {
+  logger.error(err.stack);
+}
+
+// Graceful shutdown
+process.once("SIGTERM", () => shutDown().then(() => process.exit()));
+
+// In development mode the app is launched with an IPC channel
+if (process.channel) {
+  // Prevent exiting the process in development mode
+  process.on("uncaughtException", handleError);
+  // Restart the server on code changes (see scripts/run.js)
+  process.on("message", message => {
+    if (message === "reload") {
+      reload = reload.then(() => shutDown()).then(() => {
+        Object.keys(require.cache).forEach(key => {
+          if (key.indexOf("node_modules") === -1) delete require.cache[key];
```

Comment:
why do you manually invalidate cache??? I am not sure it's a good idea..
for such restart purposes simple pkg [nodemon](https://www.npmjs.com/package/nodemon).

I think better to delegate restart init to this tool

Reply:
this approach we use for very long time for `plyo.db`, it was part of node-starter-kit for some time I believe. It worked perfect, but I need to test how it works in boilerplate. Looks like I forgot about some dependencies. I'll think about alternatives. Meanwhile @frenzzy what's you opinion on this question?

Reply:
@frenzzy reminded me how nodemon works, and it's much slower. It's not critical for small projects, but plyo.db initialization took a lot of time and we want to keep this process similar for all the projects.

Reply:
@jifeon slow? hm, interesting. didn't face such problems with it before.
just curios about circumstances where it has been slow